app/services/notify_service.rb

```diff
@@ -16,6 +17,14 @@ def call(recipient, activity)
 
   private
 
+  def visibility
+    unless !@visibility.nil? || @activity.is_a?(Follow) || @activity.is_a?(FollowRequest)
```

Comment:
How about using `return @status_visibility if defined?(@status_visibility)`? Then you can cache `nil` too.

Reply:
sounds good, i'll fix that up